src/components/Button.stories.tsx

```diff
@@ -151,10 +151,31 @@ export function ButtonWithTooltip() {
 }
 
 export function AsyncButton() {
+  const ea = <Icon icon={"cloudUpload"} />;
+  const eaInFlight = <Loader size="xs" />;
+  const label = "Upload";
+  const labelInFlight = "Uploading";
+  const icon = "archive";
+  const iconInFlight = "sunny";
+
   return (
-    <div>
-      <h2>Clicking the button will disable it for 2 seconds (while async request is in progress)</h2>
-      <Button label="Upload" onClick={async () => await new Promise((resolve) => setTimeout(resolve, 2000))} />
-    </div>
+    <>
+      <div>
+        <h2>Clicking the button will disable it for 2 seconds (while async request is in progress)</h2>
+        <Button label="Upload" onClick={async () => await new Promise((resolve) => setTimeout(resolve, 2000))} />
+      </div>
+      <div css={Css.mt4.$}>
+        <h2>Additional Props may be passed in to further customize in flight look</h2>
+        <Button
+          label={label}
+          endAdornment={ea}
+          icon={icon}
+          inFlightEndAdornment={eaInFlight}
+          inFlightIcon={iconInFlight}
+          inFlightLabel={labelInFlight}
+          onClick={async () => await new Promise((resolve) => setTimeout(resolve, 2000))}
```

Comment:
**thought:** I like this pattern a lot. But it makes me wonder if we really need to have separate props for this... like what if we just expected people to do:
```
const [save, {loading}] = useMutation();
<Button label={loading ? "Uploading..." : "Upload} ... />
```
?
Maybe having these ternary statements here isn't as nice as hiding them behind the curtain within `Button`. But I dunno... Part of me also would rather standardize on some of this... like maybe the label changes, but perhaps we _always_ show that Loader for async buttons? Other than the loader and changing the text, I'm not sure I see the benefit of the `icon` being able to change. 🤔  I think it'd be odd if the loading state for our Button was different depending on who impelments it, so standardizing on always showing that `Loader` would be better imo.

Reply:
I'm not picky about leaving `Icon` out and making `Loader` standard for anything async is verrry nice. It works for my use case 😏  

Keeping `inFlightLabel?: string" is great for a HOC. In our use-case we're passing the onClick Fn around so if we can leave out loading that's 1 less prop to define and pass down.